src/idl_parser.cpp

```diff
@@ -1234,6 +1279,58 @@ CheckedError Parser::ParseVector(const Type &type, uoffset_t *ovalue,
   return NoError();
 }
 
+CheckedError Parser::ParseArray(Value &array) {
+  std::vector<Value> stack;
+  FlatBufferBuilder builder;
+  const auto &type = array.type.VectorType();
+  auto length = array.type.fixed_length;
+
+  uoffset_t count = 0;
+  auto err = ParseVectorDelimiters(count, [&](uoffset_t &) -> CheckedError {
+    Value val;
+    val.type = type;
+    if (IsStruct(type)) {
+      ECHECK(ParseTable(*val.type.struct_def, &val.constant, nullptr));
+    } else {
+      ECHECK(ParseSingleValue(nullptr, val, false));
+    }
+    stack.push_back(val);
+    return NoError();
+  });
+  ECHECK(err);
+
+  if (length != count) return Error("Fixed-length array size is incorrect.");
+
+  for (size_t i = 0; i < count; i++) {
```

Comment:
Is it possible to use reverse iterator over `stack`?
The reverse iterator will remove `stack.pop_back()`.

Reply:
Using iterators now.